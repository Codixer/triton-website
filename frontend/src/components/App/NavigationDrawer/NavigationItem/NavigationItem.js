import React, { Component } from "react";
import styles from "./NavigationItem.scss";
import { connect } from "react-redux";
import "@material/list/dist/mdc.list.css";
import { ListItem, ListItemGraphic } from "@rmwc/list";
import { setDrawerState } from "../../../../actions/navigationActions";
import ScrollIntoView from "react-scroll-into-view";

class NavigationItem extends Component {
  render() {
    return (
      <div onClick={this.props.closeDrawer} className={styles.navItem}>
        <ScrollIntoView selector={this.props.selector} smooth alignToTop>
          <ListItem>
            <ListItemGraphic icon={this.props.icon} />
            {this.props.children}
          </ListItem>
        </ScrollIntoView>
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  closeDrawer: () => dispatch(setDrawerState(false)),
});

export default connect(
  null,
  mapDispatchToProps
)(NavigationItem);
