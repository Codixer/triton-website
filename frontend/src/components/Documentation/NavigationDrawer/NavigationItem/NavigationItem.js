import React, { Component } from "react";
import styles from "./NavigationItem.scss";
import { connect } from "react-redux";
import "@material/list/dist/mdc.list.css";
import { ListItem, ListItemGraphic } from "@rmwc/list";
import { setDrawerState } from "../../../../actions/navigationActions";
import { Link } from "react-router-dom";

class NavigationItem extends Component {
  render() {
    return (
      <div onClick={this.props.closeDrawer} className={styles.navItem}>
        <Link to={this.props.to}>
          <ListItem>
            {this.props.icon && <ListItemGraphic icon={this.props.icon} />}
            {this.props.children}
          </ListItem>
        </Link>
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  closeDrawer: () => dispatch(setDrawerState(false)),
});

export default connect(
  null,
  mapDispatchToProps
)(NavigationItem);
