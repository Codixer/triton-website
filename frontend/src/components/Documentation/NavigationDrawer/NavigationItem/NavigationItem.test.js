import React from "react";
import { shallow } from "enzyme";
import { NavigationItem } from "./NavigationItem";
import { ListItemGraphic } from "@rmwc/list";

describe("Documentation/NavigationDrawer/NavigationItem Component", () => {
  let mockCloseDrawer = jest.fn();

  let wrapper = shallow(
    <NavigationItem icon="star" to="/test" closeDrawer={mockCloseDrawer}>
      Test Item
    </NavigationItem>
  );

  it("renders icon", () => {
    expect(wrapper.contains(<ListItemGraphic icon="star" />)).toBe(true);
  });

  it("renders children", () => {
    expect(wrapper.find("ListItem").props().children).toEqual([
      <ListItemGraphic icon="star" />,
      "Test Item",
    ]);
  });

  it("passes target destination to Link", () => {
    expect(wrapper.find("Link").props().to).toBe("/test");
  });

  it("closesDrawer onclick", () => {
    wrapper.simulate("click");
    expect(mockCloseDrawer).toHaveBeenCalledTimes(1);
  });

  it("matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
