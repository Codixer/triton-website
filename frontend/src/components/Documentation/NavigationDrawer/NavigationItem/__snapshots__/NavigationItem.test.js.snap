// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Documentation/NavigationDrawer/NavigationItem Component matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavigationItem
    closeDrawer={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    icon="star"
    to="/test"
  >
    Test Item
  </NavigationItem>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Link
        replace={false}
        to="/test"
      >
        <ListItem
          ripple={true}
          tabIndex={0}
        >
          <ListItemGraphic
            icon="star"
            tag={[Function]}
          />
          Test Item
        </ListItem>
      </Link>,
      "className": undefined,
      "onClick": [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <ListItem
          ripple={true}
          tabIndex={0}
        >
          <ListItemGraphic
            icon="star"
            tag={[Function]}
          />
          Test Item
        </ListItem>,
        "replace": false,
        "to": "/test",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <ListItemGraphic
              icon="star"
              tag={[Function]}
            />,
            "Test Item",
          ],
          "ripple": true,
          "tabIndex": 0,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "icon": "star",
              "tag": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "Test Item",
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Link
          replace={false}
          to="/test"
        >
          <ListItem
            ripple={true}
            tabIndex={0}
          >
            <ListItemGraphic
              icon="star"
              tag={[Function]}
            />
            Test Item
          </ListItem>
        </Link>,
        "className": undefined,
        "onClick": [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <ListItem
            ripple={true}
            tabIndex={0}
          >
            <ListItemGraphic
              icon="star"
              tag={[Function]}
            />
            Test Item
          </ListItem>,
          "replace": false,
          "to": "/test",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <ListItemGraphic
                icon="star"
                tag={[Function]}
              />,
              "Test Item",
            ],
            "ripple": true,
            "tabIndex": 0,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "icon": "star",
                "tag": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "Test Item",
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
